### PR TITLE
manager: fix TLS certs for tiproxy

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve?

1. Correctly set tiproxy certs for new tiproxy version. It is set to tidb certs directly to share same certs as tidb to conform components using tidb certs to connect tiproxy. It also simplifies the configruation of tiproxy, now, no need for configs of tiproxy TLS by default.
2. Use `TiDBClientTLSSecretName(tc.Name, xx)` to simply if statement

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
share same certs for TiProxy and TiDB
```

### CODE REVIEW

File: pkg/util/util.go [hunk 1]
```diff
@@ -193,7 +193,10 @@ func ClusterTLSSecretName(tcName, component string) string {
 	return fmt.Sprintf("%s-%s-cluster-secret", tcName, component)
 }
 
-func TiDBClientTLSSecretName(tcName string) string {
+func TiDBClientTLSSecretName(tcName string, secretName *string) string {
+	if secretName != nil {
+		tcName = *secretName
+	}
 	return fmt.Sprintf("%s-tidb-client-secret", tcName)
 }
```
Comment (KanShiori) [inline pkg/util/util.go:201]: When `secretName` is set, it should return `secretName`, right?

Comment (xhebox) [reply pkg/util/util.go]: Yes. I missed that.

**Review by KanShiori — APPROVED** [review]

**Review by csuzhangxc — APPROVED** [review]

**Comment by xhebox** [comment]
/hold

There is another bug. I'll try to fix it in this PR.

**Pull request merged**